packages/openapi-generator/src/parser/schema-naming.ts

```diff
@@ -36,39 +29,52 @@ function makeSchemaNameValid(name: string): string {
  * @param options Parser options.
  * @returns A list of strings which are possible for schema names.
  */
-export function ensureUValidSchemaNames(
+export function ensureValidSchemaNames(
   names: string[],
   options: ParserOptions
 ): string[] {
-  const nonValidNames = names.filter(name => !isValidSchemaName(name));
-
-  if (nonValidNames.length > 0) {
-    if (options.strictNaming) {
-      throw new Error(getNonValidNameErrorMessages(nonValidNames));
-    }
-    return names.map(name => makeSchemaNameValid(name));
+  if (options.strictNaming) {
+    validateSchemaNames(names);
   }
-  return names;
+
+  return names.map(name => makeSchemaNameValid(name));
 }
 
-function getNonValidNameErrorMessages(nonValidSchemaNames: string[]) {
-  const header =
-    "Your OpenApi definition contains the invalid schema names. The SDK generator can adjust such names automatically if you disable 'strictNaming' or you adjust the schema. The errors are:";
-  const errors = nonValidSchemaNames.map(nonValidSchemaName =>
-    getErrorMessage(nonValidSchemaName)
+function validateSchemaNames(names: string[]): void {
+  const namesStartingWithNumber = names.filter(name =>
+    name.match(startsWithNumberRegex)
+  );
+  const namesContainingIllegalChars = names.filter(name =>
```

Comment:
```suggestion
  const namesContainingIllegalChars = names.some(name =>
```

I think the `length` is not relevant so `.some` would be the better fit instead of `.filter`

Reply:
Nevermind, it's needed for the errors...